### PR TITLE
Rroux editor

### PR DESCRIPTION
**Pull request by rroux-gpsw**
Code needed for the editor.

### CODE REVIEW

**Comment by sw-team-release-gpsw** [comment]
[CLA assistant check](https://cla.gopro.com/sign) <br/>All committers have signed the CLA.

File: src/math/Math.js [hunk 1]
```diff
@@ -259,21 +259,28 @@ FORGE.Math.sphericalToCartesian = function(radius, theta, phi)
  * @param {number} x - x
  * @param {number} y - y
  * @param {number} z - z
+ * @param {string} [unit=radian] - The unit used to return spherical
```
Comment (aberthet-gpsw) [inline src/math/Math.js]: Can't we add this in the `sphericalToCartesian` method as well ?

File: src/math/Math.js [hunk 2]
```diff
@@ -259,21 +259,28 @@ FORGE.Math.sphericalToCartesian = function(radius, theta, phi)
...
     if (res.radius === 0)
     {
-        return { radius: 0, theta: 0, phi: 0 }
+        return { radius: 0, theta: 0, phi: 0 };
     }
 
     res.phi = Math.asin(y / res.radius);
     res.theta = Math.atan2(x, -z || 0); // we want to avoid -z = -0
 
+    if(typeof unit === "string" && unit.toLowerCase().substring(0, 3) === "deg")
```
Comment (aberthet-gpsw) [inline src/math/Math.js]: Not a fan of this condition, can't we have a constant for this ? The `.toLowerCase().substring(0, 3)` is weird.

File: src/render/RenderManager.js [hunk 3]
```diff
@@ -1174,6 +1203,21 @@ Object.defineProperty(FORGE.RenderManager.prototype, "backgroundReady",
 });
 
 /**
+ * Get the objects
```
Comment (aberthet-gpsw) [inline src/render/RenderManager.js]: Please provide a more complete description, this one is really unclear.

File: reference/hotspot/hotspot.json [hunk 4]
```diff
@@ -94,7 +94,5 @@
         "states": {
             "$ref": "hotspot-states"
         }
-    },
-
-    "required": [ "uid", "tags", "type", "visible", "transform" ]
```
Comment (rroux-gpsw) [inline reference/hotspot/hotspot.json]: Because none are required :)

File: src/hotspots/Hotspot3D.js [hunk 5]
```diff
@@ -34,6 +34,14 @@ FORGE.Hotspot3D = function(viewer, config)
     this._transform = null;
 
     /**
+     * HotspotGeometry object.
+     * @name FORGE.Hotspot3D._geometry
```
Comment (rroux-gpsw) [inline src/hotspots/Hotspot3D.js]: well seen

File: src/math/Math.js (hunk 1 above)
Comment (rroux-gpsw) [inline src/math/Math.js]: What unit would you like to add to the return value of sphericalToCartesian ? Cartesian result is world unit only, am I right?

Comment (aberthet-gpsw) [inline src/math/Math.js]: Yes, but arguments can be either radian or degrees.

File: src/math/Math.js (hunk 2 above)
Comment (rroux-gpsw) [inline src/math/Math.js]: Yes I thought about it but asked myself if it not a bit too much. But ok, I do agree I'll add something like Math.units.RADIAN and Math.units.DEGREE

Comment (rroux-gpsw) [inline src/math/Math.js]: Oh, it already exists 

FORGE.Math.DEGREES = "degrees";

Comment (rroux-gpsw) [inline src/math/Math.js]: A unit name is not a singular word?

Comment (aberthet-gpsw) [inline src/math/Math.js]: Yes indeed, `DEGREES` can be moved to `DEGREE`.

Comment (rroux-gpsw) [inline src/math/Math.js]: BTW about this condition my first thought was to be able to accept "deg" "degree" or "degrees". I couldn't make a choice haha!

File: src/render/RenderManager.js (hunk 3 above)
Comment (rroux-gpsw) [inline src/render/RenderManager.js]: ok : Get the FORGE.ObjectRenderer instance

File: src/math/Math.js (hunk 1 above)
Comment (rroux-gpsw) [inline src/math/Math.js]: Yes of course, but I can't call this is the same approach. I'll do this

Comment (rroux-gpsw) [inline src/math/Math.js]: Done :)

**Pull request merged**